### PR TITLE
GH-14784: [Dev] Add possibility to autoassign on GitHub issue comment

### PR DESCRIPTION
**Pull request by raulcd**
This PR will allow a user to type `take` in an issue comment and the issue will automatically be assigned to them.

### CODE REVIEW

**Comment by raulcd** [comment]
I have tested this workflow on my fork: https://github.com/raulcd/arrow/issues/60#issuecomment-1332289389

**Review by jorisvandenbossche — APPROVED** [review]
Looks good!

File: .github/workflows/comment_bot.yml [hunk 1]
```diff
@@ -174,3 +174,18 @@ jobs:
...
           args: "--force"
+
+  issue_assign:
+    name: "Assign issue"
+    permissions:
+      issues: write
+    if: github.event.comment.body == 'take'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```
Comment (assignUser) [inline .github/workflows/comment_bot.yml:185]: We can remove this checkout by making the .js file into an action, that would be more idiomatic for gha.

Comment (assignUser) [reply .github/workflows/comment_bot.yml:185]: but I can work on that as a follow up too :)

Comment (raulcd) [reply .github/workflows/comment_bot.yml]: Thanks for the comment, it made me realise I could simplify it a lot as when I started I thought I might need more code for it. The action is not needed though.

**Review by assignUser — APPROVED** [review]
Nice!

**Review by kou — APPROVED** [review]
+1

**Pull request merged**